File: src/structure/derived/derivedDeclaration.ts
import {ModuleDeclaration} from "typescript";
import {StructureItem} from "../structure.js";

export interface DerivedDeclaration {
    sourceFileName: string,
    namespace: ModuleDeclaration | undefined,
    fileName: string,
    body: string,
}

export interface DerivedDeclarationStructureItem extends StructureItem {
    body: string,
}

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/structure/derived/DerivedDeclaration.kt
package io.github.sgrishchenko.karakum.structure.derived

import io.github.sgrishchenko.karakum.structure.StructureItem
import js.objects.JsPlainObject
import typescript.ModuleDeclaration

@JsPlainObject
external interface DerivedDeclaration {
    val sourceFileName: String
    val namespace: ModuleDeclaration?
    val fileName: String
    val body: String
}

@JsPlainObject
external interface DerivedDeclarationStructureItem : StructureItem {
    val body: String
}
